mito-ai/src/Extensions/AiChat/ChatTaskpane.tsx

```diff
@@ -473,59 +478,148 @@ const ChatTaskpane: React.FC<IChatTaskpaneProps> = ({
-                    aiResponse.items.forEach((item: any) => {
-                        newChatHistoryManager.addAIMessageFromResponse(
-                            item.content || '',
-                            completionRequest.metadata.promptType
-                        );
-                    });
-                    setChatHistoryManager(newChatHistoryManager);
+                    const content = aiResponse.items[0]?.content ?? '';
+
+                    if (completionRequest.metadata.promptType === 'agent:execution' || completionRequest.metadata.promptType === 'agent:autoErrorFixup') {
+                        // Agent:Execution prompts return a CellUpdate object that we need to parse
+                        const agentResponse: AgentResponse = JSON.parse(content)
+                        newChatHistoryManager.addAIMessageFromAgentResponse(agentResponse)
+                    } else {
+                        // For all other prompt types, we can just add the content to the chat history
+                        aiResponse.items.forEach((item: any) => {
+                            newChatHistoryManager.addAIMessageFromResponse(
+                                item.content || '',
+                                completionRequest.metadata.promptType
+                            );
+                        });
+                        setChatHistoryManager(newChatHistoryManager);
+                    }
                 }
+            } catch (error) {
+                addAIMessageFromResponseAndUpdateState(
+                    (error as any).title ? (error as any).title : `${error}`,
+                    'chat',
+                    newChatHistoryManager,
+                    false
+                );
+                addAIMessageFromResponseAndUpdateState(
+                    (error as any).hint ? (error as any).hint : `${error}`,
+                    completionRequest.metadata.promptType,
+                    newChatHistoryManager,
+                    true
+                )
+            } finally {
+                // Reset states to allow future messages to show the "Apply" button
+                setCodeReviewStatus('chatPreview');
+                setLoadingAIResponse(false);
             }
-        } catch (error) {
-            addAIMessageFromResponseAndUpdateState(
-                (error as any).title ? (error as any).title : `${error}`,
-                'chat',
-                newChatHistoryManager,
-                false
-            );
-            addAIMessageFromResponseAndUpdateState(
-                (error as any).hint ? (error as any).hint : `${error}`,
-                completionRequest.metadata.promptType,
-                newChatHistoryManager,
-                true
-            )
-        } finally {
-            // Reset states to allow future messages to show the "Apply" button
-            setCodeReviewStatus('chatPreview');
+        } else {
```

Comment:
The giant try statement here and above are exactly the same (I just did a CMD+F). So can't we just change this up so that we have

```typescript
if (streaming) {
    // set up stream handling stuff
}

try {
    // send the message and do all of the handling
}
//... etc

```

Reply:
Actually, see other comment, I think we don't need all of this stuff for the streaming case